scripts/lib/CIME/XML/grids.py

```diff
@@ -57,7 +57,7 @@ def get_grid_info(self, name, compset):
             name = levmatch.group(1)+levmatch.group(3)
 
         #mechanism to specify lnd levels
-        levmatch = re.match(r"(.*_)([^_]+)z(\d+)(.*)$", name)
+        levmatch = re.match(r"(.*_)([^_]+)z(\d+)(_[^m].*)$", name)
```

Comment:
No but at the moment, there is no way to specify land levels if it is not present. For example:
`ne30z192_ne30` always specifies atmosphere levels. If you want to specify land levels, you need a tri-grid short name which means group 4 will be present.